### PR TITLE
[SERF-3458] Introduce Statsig package

### PR DESCRIPTION
**Pull request by laynax**
## Description

This PR introduces Statsig to leverage the functionality of `experimentation` and `feature flag`.
It includes:
- A new `statsig` package along with its configuration.
- Defined `ToContext` and `Extract` for both `experiment` and `featureFlag`.
- Introduced middlewares and interceptors to inject experiment and featureFlag into the context to be used in middleware.
- All include unit tests

## Testing considerations

Tested on [development](https://github.com/scribd/go-chassis/actions/runs/10077016940)

## Checklist

- [x] Prefixed the PR title with the JIRA ticket code 
- [x] Performed simple, atomic commits with [good commit messages][commit messages]
- [x] Verified that the commit history is linear and commits are squashed as necessary
- [x] Thoroughly tested the changes in `development` and/or `staging`
- [ ] Updated the `README.md` as necessary

## Related links

* [SERF-3458](https://scribdjira.atlassian.net/browse/SERF-3458)
* https://pkg.go.dev/github.com/statsig-io/go-sdk
* https://docs.statsig.com/
* https://github.com/scribd/go-chassis/pull/177
* https://github.com/scribd/terraform-services-go-chassis/pull/85

[SERF-3458]: https://scribdjira.atlassian.net/browse/SERF-3458?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/context/statsig/context.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/scribd/go-sdk/pkg/statsig"
+)
+
+type (
+	ctxStatsigExperimentMarker  struct{ fn statsig.GetExperimentFunc }
+	ctxStatsigFeatureFlagMarker struct{ fn statsig.GetFeatureFlagFunc }
+)
+
+var (
+	ctxStatsigExperminetKey  = &ctxStatsigExperimentMarker{}
```
Comment (fotos) [inline pkg/context/statsig/context.go:16]:
```suggestion
	ctxStatsigExperimentKey  = &ctxStatsigExperimentMarker{}
```

**Review by fotos — COMMENTED** [review]
Did a quick pass and left a few remarks. 💬

**Review by laynax — COMMENTED** [review]
@fotos Nice catches 👍 
Fixed the typos and added 2 new test files ⛏️

File: pkg/interceptors/request_id_test.go
Comment (fotos) [inline pkg/interceptors/request_id_test.go:1]: I'm not sure I understand this change as part of this PR, but OK. 😊

File: pkg/context/statsig/context.go
Comment (fotos) [inline pkg/context/statsig/context.go:1]: Maybe we should add [tests for this file](https://github.com/scribd/go-sdk/blob/0dda53000131e693335248006135a77629564a5a/pkg/context/requestid/context_test.go) too? 🤔

**Review by fotos — DISMISSED** [review]
:+1: LGTM 🚀 

Approved the PR – left 2 more (nit) remarks.

File: pkg/interceptors/request_id_test.go
Comment (laynax) [reply pkg/interceptors/request_id_test.go:1]: Linter failed. Apparently, hardcoded strings are allowed to be repeated only 3 times.

File: pkg/context/statsig/context.go
Comment (laynax) [reply pkg/context/statsig/context.go:1]: Done.

**Review by fotos — DISMISSED** [review]
:+1: LGTM :rocket: 

Does the documentation need an update too?

**Review by fotos — DISMISSED** [review]
:+1: LGTM 🚀

**Review by fotos — APPROVED** [review]
:+1: LGTM 🚀

**Pull request merged**